pkg/registry/directoryGraphLoader.go

```diff
@@ -0,0 +1,223 @@
+package registry
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+
+	"github.com/blang/semver"
+)
+
+type DirGraphLoader struct {
+	PackageDir           string
+	CsvNameAndReplaceMap map[string]csvReplaces
+	SortedCSVs           csvs // only contains bundles with version field which will be considered for skip range.
+}
+
+type csvReplaces struct {
+	replaces  []string
+	skipRange semver.Range
+}
+
+type csv struct {
+	name    string
+	version semver.Version
+}
+
+type csvs []csv
+
+func (c csvs) Len() int           { return len(c) }
+func (c csvs) Less(i, j int) bool { return c[i].version.LT(c[j].version) }
+func (c csvs) Swap(i, j int)      { c[i], c[j] = c[j], c[i] }
+
+// NewPackageGraphLoaderFromDir takes the root directory of the package in the file system.
+func NewPackageGraphLoaderFromDir(packageDir string) (*DirGraphLoader, error) {
+	_, err := ioutil.ReadDir(packageDir)
+	if err != nil {
+		return nil, fmt.Errorf("error reading from %s directory, %v", packageDir, err)
+	}
+
+	loader := DirGraphLoader{
+		PackageDir: packageDir,
+	}
+
+	err = loader.loadBundleCsvPathMap(packageDir)
+	if err != nil {
+		return nil, fmt.Errorf("error geting CSVs from all bundles in the package directory, %v", err)
+	}
+
+	loader.loadReplaces()
```

Comment:
Why isn't this function just part of the `loader.Generate()`? It seems like this will make it harder to mock if some of the implementation is inside the constructor.